packages/wonder-blocks-dropdown/components/dropdown-core.js

```diff
@@ -65,22 +79,59 @@ export default class DropdownCore extends React.Component<DropdownCoreProps> {
     }
 
     componentDidMount() {
+        this.updateEventListeners();
+    }
+
+    componentDidUpdate(prevProps: DropdownCoreProps) {
+        if (prevProps.open !== this.props.open) {
+            this.updateEventListeners();
+        }
+
+        // If the menu just changed from closed to open, and if we haven't
+        // found the correct menuWidth yet. Only applies to right aligned.
+        if (this.props.alignment === "right" && !prevProps.open && this.props.open && !this.menuWidth) {
+            // There is a timeout because if we call getBoundingClientRect
+            // right away, it returns the value of the document width instead
+            // of the width of the just-mounted component, the menu.
+            requestAnimationFrame(() => {
+                if (this.menuElement) {
```

Comment:
It seems like this check can be moved above to avoid calling this function in the first place, potentially?

Reply:
The check is to satisfy Flow, but in this method `this.menuElement` should be defined.

Reply:
Since the context of the callback is different, Flow doesn't allow me to move it above.